KerbalVR_Mod/KerbalVR-MAS/MASLever.cs

```diff
@@ -0,0 +1,153 @@
+﻿using AvionicsSystems;
+using KerbalVR;
+using KerbalVR.InternalModules;
+using KerbalVR.IVAAdaptors;
+using System;
+
+namespace KerbalVR_MAS
+{
+	public class MASLever : IVALever
+	{
+		public static MASLever TryConstruct(VRLever vrLever)
+		{
+			MASComponent component = vrLever.GetComponent<MASComponent>(); 
+
+			if (component)
+			{
+				return new MASLever(component, vrLever);
+			}
+
+			return null;
+		}
+
+		VRLever lever;
+		MASComponentAnimation componentAnimation;
+		MASComponentAnimationPlayer componentAnimationPlayer;
+		MASComponentRotation componentRotation;
+		MASFlightComputer flightComputer;
+
+		// cache it because some MAS functions won't update value instantly
+		int lastStep;
+
+		public MASLever(MASComponent component, VRLever vrLever)
+		{
+			lever = vrLever;
+
+			foreach (IMASSubComponent action in component.actions)
+			{
+				if (action is MASComponentAnimation componentAnimation)
+				{
+					this.componentAnimation = componentAnimation;
+				}
+				else if (action is MASComponentAnimationPlayer componentAnimationPlayer)
+				{
+					this.componentAnimationPlayer = componentAnimationPlayer;
+				}
+				else if (action is MASComponentRotation componentRotation)
+				{
+					this.componentRotation = componentRotation;
+				}
+			}
+
+			flightComputer = vrLever.part.GetComponent<MASFlightComputer>();
+
+			lastStep = RefreshState();
+		}
+
+		public override void SetStep(int stepId)
+		{
+			switch (lever.handler)
+			{
+				case "Gear":
+					flightComputer.fcProxy.SetGear(stepId == 1);
+					break;
+				case "Spoiler":
+					flightComputer.farProxy.SetSpoilers(stepId == 1);
+					break;
+				case "Flap":
+					int currentSetting = (int)Math.Round(flightComputer.farProxy.GetFlapSetting());
+					if (currentSetting >= 0)
+					{
+						int delta = stepId - currentSetting;
+						if (delta < 0)
+						{
+							for (int i = 0; i > delta; --i)
+							{
+								flightComputer.farProxy.DecreaseFlapSetting();
+							}
+						}
+						else if (delta > 0)
+						{
+							for (int i = 0; i < delta; ++i)
+							{
+								flightComputer.farProxy.IncreaseFlapSetting();
+							}
+						}
+					}
+					break;
+				default:
+					Utils.LogWarning($"Unknown lever handler {lever.handler}");
```

Comment:
should include the prop name here if possible (you should be able to get to it through the VRLever)